File: sieve/src/main/java/local/example/outcome/OutcomeResource.java
package local.example.outcome;

import io.smallrye.mutiny.Uni;
import local.example.outcome.model.Sieve;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Path("/outcome")
public class OutcomeResource {

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Uni<Sieve> read() {
        return null;
    }
}

File: REST/harmonic-mean/src/main/kotlin/local/example/outcome/OutcomeResource.kt
package local.example.outcome

import local.example.outcome.model.HarmonicMean
import java.util.*
import javax.ws.rs.*
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/outcome")
class OutcomeResource {

    private val harmonicMeans = Collections.newSetFromMap(
        Collections.synchronizedMap(
            LinkedHashMap<HarmonicMean, Boolean>()
        )
    )

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    fun read(): Response? {
        return if (
            harmonicMeans.isEmpty()
        ) Response.ok().build() else Response.ok(harmonicMeans).build()
    }

    @POST
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    fun compute(harmonic: HarmonicMean): Set<HarmonicMean?>? {
        harmonic.setHarmonicMu()
        harmonicMeans.add(harmonic)
        return harmonicMeans
    }
}
